pkg/apb/executor.go

```diff
@@ -0,0 +1,124 @@
+package apb
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	logging "github.com/op/go-logging"
+	"github.com/openshift/ansible-service-broker/pkg/clients"
+	"github.com/pborman/uuid"
+	"k8s.io/kubernetes/pkg/api/v1"
+)
+
+////////////////////////////////////////////////////////////
+// TODO (Erik): Need to refactor to get rid of 80% of the usages
+// of this. It's all over the place and doesn't need to be.
+// Think it's going to turn into the natural place for the
+// configurable svcaccount cert/token location and openshift target
+//
+// I'm planning on submitting a PR to fix this. Yell at me if
+// that hasn't been done. :)
+////////////////////////////////////////////////////////////
+
+// ClusterConfig - DEPRECATED Configuration struct for clusters
+type ClusterConfig struct {
+	InCluster bool
+	Target    string
+	User      string
+	Password  string `yaml:"pass"`
+}
+
+////////////////////////////////////////////////////////////
+
+// ExecuteApb - Runs an APB Action with a provided set of inputs
+func ExecuteApb(
+	action string,
+	clusterConfig ClusterConfig,
+	spec *Spec,
+	context *Context,
+	p *Parameters,
+	log *logging.Logger,
+) (string, error) {
+	extraVars, err := createExtraVars(context, p)
+
+	if err != nil {
+		return "", err
+	}
+
+	log.Debug("ExecutingApb:")
+	log.Debug("name:[ %s ]", spec.Name)
+	log.Debug("image:[ %s ]", spec.Image)
+	log.Debug("action:[ %s ]", action)
+
+	// It's a critical error if a Namespace is not provided to the
+	// broker because its required to know where to execute the pods and
+	// sandbox them based on that Namespace. Should fail fast and loud,
+	// with controlled error handling.
+	if context.Namespace == "" {
+		errStr := "Namespace not found within request context. Cannot perform requested " + action
+		log.Error(errStr)
+		return "", errors.New(errStr)
```

Comment:
+1. I don't like the length and normally endeavor to deal with it, but this is not a big deal.